### PR TITLE
DM-49949: Add astropy.table data ID join method to Query.

### PR DESCRIPTION
**Pull request by TallJimbo**
## Checklist

- [x] ran Jenkins
- [x] added a release note for user-visible changes to `doc/changes`
- [ ] (if changing dimensions.yaml) make a copy of dimensions.yaml in `configs/old_dimensions`

### CODE REVIEW

File: python/lsst/daf/butler/queries/_query.py [hunk 1]
```diff
@@ -618,6 +621,50 @@ def join_data_coordinates(self, iterable: Iterable[DataCoordinate]) -> Query:
...
+            but is fully constrained to a literal by a previous `where` call, a
+            constant-valued column will be added.
+
+        Returns
+        -------
+        query : `Query`
+            A new query object with the data IDs joined in.
+        """
+        if not len(table):
+            raise InvalidQueryError("Cannot upload an empty data coordinate set.")
```
Comment (MichelleGower) [inline python/lsst/daf/butler/queries/_query.py:642]: Is it easy to add a unit test for this handling of an empty table?

**Review by MichelleGower — APPROVED** [review]
Looks good.  Pointed out one place where hope the unit test would be easy.  Merge approved.
